src/e2e/java/teammates/e2e/cases/e2e/StudentFeedbackResultsPageE2ETest.java

```diff
@@ -0,0 +1,278 @@
+package teammates.e2e.cases.e2e;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.FeedbackParticipantType;
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.e2e.pageobjects.AppPage;
+import teammates.e2e.pageobjects.StudentFeedbackResultsPage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#STUDENT_SESSION_RESULTS_PAGE}.
+ */
+public class StudentFeedbackResultsPageE2ETest extends BaseE2ETestCase {
+    private StudentFeedbackResultsPage resultsPage;
+    private FeedbackSessionAttributes openSession;
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/StudentFeedbackResultsPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        openSession = testData.feedbackSessions.get("Open Session");
+    }
+
+    @Test
+    public void testAll() {
+
+        ______TS("unregistered student can access results");
+        StudentAttributes unregistered = testData.students.get("Unregistered");
+        AppUrl url = createUrl(Const.WebPageURIs.SESSION_RESULTS_PAGE)
+                .withCourseId(unregistered.course)
+                .withStudentEmail(unregistered.email)
+                .withSessionName(openSession.getFeedbackSessionName())
+                .withRegistrationKey(getKeyForStudent(unregistered));
+        logout();
+        resultsPage = AppPage.getNewPageInstance(browser, url, StudentFeedbackResultsPage.class);
+
+        resultsPage.verifyFeedbackSessionDetails(testData.feedbackSessions.get("Open Session"));
```

Comment:
I think this one can benefit from checking the questions loaded as well. However unlikely it might be, we don't want a situation whereby the session details are loaded but the questions are not to pass here.